src/parser/Statement.ts

```diff
@@ -38,6 +64,9 @@ export interface Statement {
 
     /** The starting and ending location of the expression. */
     location: Location;
+
+    /** The type of statement. */
+    type: string;
```

Comment:
Didn't think about that -- great idea 😄

Reply:
This didn't _quite_ work out as cleanly as I hoped, because `AstNode` and `Statement`/`Expression` have no knowledge of each other. So Typescript didn't like it when I tried to cast one to the other. But I was able to do `statement as unknown as AstNode`, if that's better 🤷‍♂️